actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -355,7 +355,7 @@ def split_to(to)
 
           def add_controller_module(controller, modyoule)
             if modyoule && !controller.is_a?(Regexp)
-              if %r{\A/}.match?(controller)
+              if !controller.nil? && %r{\A/}.match?(controller)
```

Comment:
`if controller && controller.start_with?('/')` would be more readable here.

Reply:
Using `start_with?` here should have been enforced by RuboCop in https://github.com/rails/rails/pull/32695, but wasn't due to a bug (https://github.com/rubocop-hq/rubocop/pull/7464).

In any case, I think this patch should only contain changes required for compatibility with https://github.com/ruby/ruby/pull/1506, no matter how tempting adjacent refactorings may be.

Reply:
Using `start_with?` here should have been enforced by RuboCop in https://github.com/rails/rails/pull/32695, but wasn't due to a bug (https://github.com/rubocop-hq/rubocop/pull/7464).

In any case, I think this patch should only contain changes required for compatibility with https://github.com/ruby/ruby/pull/1506, no matter how tempting adjacent refactorings may be.

Reply:
Using `start_with?` here should have been enforced by RuboCop in https://github.com/rails/rails/pull/32695, but wasn't due to a bug (https://github.com/rubocop-hq/rubocop/pull/7464).

In any case, I think this patch should only contain changes required for compatibility with https://github.com/ruby/ruby/pull/1506, no matter how tempting adjacent refactorings may be.

Reply:
Good point, it's probably safer to just directly fix the compatibility with ruby-head in this PR.